pyrf/gui/controller.py

```diff
@@ -586,3 +601,83 @@ def user_xrange_changed(self, start, stop):
 
     def applying_user_xrange(self):
         return self._applying_user_xrange
+
+    def save_settings(self, dir):
+        cfgfile = open(dir,'w')
+        config = ConfigParser.SafeConfigParser()
+        state = self._state.to_json_object()
+
+        config.add_section('device_options')
+        for s in state:
+            if 'dict' in str(type(state[s])):
```

Comment:
Oh, I see what you're doing. trying to avoid tying the shape of the state structure to the config code. Yeah, that's reasonable.